docs/secrets.md

```diff
@@ -2,27 +2,39 @@
 When executing jobs through MLRun, the code might need access to specific secrets, for example to access data 
 residing on a data-store that requires credentials (such as a private S3 bucket), or many other similar needs.
 
-MLRun provides some facilities that allow handling secrets and passing those secrets to execution jobs. However, it's 
+MLRun provides some facilities that allow handling secrets and passing those secrets to execution jobs. It's 
 important to understand how these facilities work, as this has implications on the level of security they provide
 and how much exposure they create for your secrets.
 
 **In this section**
 - [Overview](#overview)
-- [Secret providers](#secret-propviders)
-   - [Inline](#inline)
-   - [Environment](#environment)
-   - [File](#file)
-   - [Kubernetes](#kubernetes)
-   - [Azure Vault](#azure-vault)
-
+- [MLRun-managed secrets](#mlrun-managed-secrets)
+  - [Secret providers](#secret-providers)
+     - [Kubernetes project secrets](#kubernetes-project-secrets)
+     - [Azure Vault](#azure-vault)
+     - [Demo/Development secret providers](#demo-development-secret-providers)
+- [Externally managed secrets](#externally-managed-secrets)
+  - [Mapping secrets to environment](#mapping-secrets-to-environment)
+  - [Mapping secrets as files](#mapping-secrets-as-files)
 
 ## Overview
+There are two main use-cases for providing secrets to an MLRun job. These are:
+
+1. [Use MLRun-managed secrets](#mlrun-managed-secrets). This is a flow that enables the MLRun user (for example a 
+data scientist or engineer) to create and use secrets through interfaces that MLRun implements and manages.
+2. [Create secrets externally](#externally-managed-secrets) to MLRun using a Kubernetes secret or some other secret 
+management framework (such as Azure vault), and utilize these secrets from within MLRun to enrich execution jobs. For 
+example, the secrets are created and managed by an IT admin and the data-scientist only accesses them.
+
+The following sections will cover the details of those two use-cases.
```

Comment:
present tense, delete "will"
The following sections cover the details of those two use-cases.